shipment_advice/models/stock_move.py

```diff
@@ -17,3 +17,13 @@ class StockMove(models.Model):
     def _plan_in_shipment(self, shipment_advice):
         """Plan the moves into the given shipment advice."""
         self.shipment_advice_id = shipment_advice
+
+    def _action_done(self, cancel_backorder=False):
+        res = super()._action_done(cancel_backorder=cancel_backorder)
+        for rec in self:
+            if rec.shipment_advice_id.shipment_type == "incoming":
+                auto_validate_enabled = self.env['ir.config_parameter'].sudo().get_param(
+                    'shipment_advice.shipment_advice_auto_validate', default=False)
+                if auto_validate_enabled:
+                    rec.shipment_advice_id._action_done()
```

Comment:
You have to care about the result of the super call, this is returning the moves which are actually done. 
As well the shipment advice should only be closed if every move is done. There call your new create method `validate_when_fully_done`

```suggestion
        res = super()._action_done(cancel_backorder=cancel_backorder)
        shipment_advices = res.shipment_advice_id
        for shipment_advice in shipment_advices:
            if shipment_advice.shipment_type != "incoming" or shipment_advice.company_id.shipment_advice_auto_validate:
                continue
            shipment_advice.validate_when_fully_done()
```